helix-view/src/theme.rs

```diff
@@ -70,6 +97,58 @@ impl Loader {
             .unwrap_or_default()
     }
 
+    // merge one theme into the parent theme
+    fn merge_themes(&self, parent_theme_toml: Value, theme_toml: Value) -> Value {
+        let parent_palette = parent_theme_toml.get("palette");
+        let palette = theme_toml.get("palette");
+
+        // handle the table seperately since it needs a `merge_depth` of 2
+        // this would conflict with the rest of the theme merge strategy
+        let palette_values = if let Some(parent_palette) = parent_palette {
+            if let Some(palette) = palette {
+                merge_toml_values(parent_palette.clone(), palette.clone(), 2)
+            } else {
+                parent_palette.clone()
+            }
+        } else {
+            if let Some(palette) = palette {
+                palette.clone()
+            } else {
+                Map::new().into()
+            }
+        };
+
+        // add the palette correctly as nested table
+        let mut palette = Map::new();
+        palette.insert(String::from("palette"), palette_values);
+
+        // merge the theme into the parent theme
+        let theme = merge_toml_values(parent_theme_toml, theme_toml, 1);
+        // merge the before specially handled palette into the theme
+        merge_toml_values(theme, palette.into(), 1)
+    }
+
+    // Loads the theme data as `toml::Value` first from the user_dir then in default_dir
+    fn load_toml(&self, path: PathBuf) -> Result<toml::Value, anyhow::Error> {
```

Comment:
In a few places there's this `Result<toml::Value, anyhow::Error>` - you can `use anyhow::Result` to bring in the anyhow result which is a little cleaner: `Result<toml::Value>`

Reply:
changed all the occurences to the `anyhow::Result` type